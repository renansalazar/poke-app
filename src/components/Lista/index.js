import { usePokemones } from '../../hooks/usePokemon'
import Pokemon from '../Pokemon'
import { Link } from "react-router-dom";

const Lista = () => {

  const { isLoading, error, data, fetchNextPage } = usePokemones('pokemones')
  if(isLoading){
    return <div>Loading...</div>
  }

  if(error){
    return <div>Error Error...</div>
  }

  const handleLoadMore = () => {
    fetchNextPage()
  }

  return <div className='containerDetalle' data-testid='listaContent'>
    <div className='containerLista'>
      {
        data.pages.map(p=>{
          return p.results.map(poke=>{
            return (
              <Link to={"/detalle/" + poke.name}>
                <Pokemon data={poke} />
              </Link>
            )
          })
        })
      }
    </div>
    <div className='containerButtonLoadMore'>
      <button className='buttonLoadMore' onClick={handleLoadMore}>Cargar m√°s...</button>
    </div>
  </div>
}

export default Lista