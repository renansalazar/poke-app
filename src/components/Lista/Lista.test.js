import React from 'react';
import {render} from '@testing-library/react';
import Lista from './';
import { QueryClient, QueryClientProvider } from 'react-query';

describe('List component', () => {
  test('first call to list', () => {
    const queryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <Lista />
      </QueryClientProvider>)
    
    expect(container.firstChild).toMatchSnapshot()
  });
});