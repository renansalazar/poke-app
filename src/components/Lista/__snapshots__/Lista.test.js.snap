// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component first call to list 1`] = `
<div
  id="loading"
>
  Loading...
</div>
`;
